recipes/openldap/all/conanfile.py

```diff
@@ -1,110 +1,117 @@
+        ]
         # Need to link to -pthread instead of -lpthread for gcc 8 shared=True
         # on CI job. Otherwise, linking fails.
-        self._autotools.libs.remove("pthread")
-        self._configure_vars["LIBS"] = self._configure_vars["LIBS"].replace(
-            "-lpthread", "-pthread")
+        # tc.libs.remove("pthread")
+        # self._configure_vars["LIBS"] = self._configure_vars["LIBS"].replace("-lpthread", "-pthread")
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        tc.generate()
 
-        self._autotools.configure(
-            args=configure_args,
-            configure_dir=self._source_subfolder,
-            vars=self._configure_vars)
-        return self._autotools
+    def _patch_sources(self):
+        replace_in_file(self, os.path.join(self.source_folder, "configure"),
+                        "WITH_SYSTEMD=no\nsystemdsystemunitdir=", "WITH_SYSTEMD=no")
 
     def build(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-        autotools = self._configure_autotools()
-
-        autotools.make(vars=self._configure_vars)
+        self._patch_sources()
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
-        autotools.install(vars=self._configure_vars)
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy("COPYRIGHT", dst="licenses", src=self._source_subfolder)
-        for folder in ["var", "share", "etc", "lib/pkgconfig", "res"]:
-            tools.rmdir(os.path.join(self.package_folder, folder))
-        tools.remove_files_by_mask(
-            os.path.join(
-                self.package_folder,
-                "lib"),
-            "*.la")
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            autotools.install()
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "COPYRIGHT", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        if os.path.exists(os.path.join(self.source_folder, "libexec")):
+            move_folder_contents(self, os.path.join(self.source_folder, "libexec"),
+                                 os.path.join(self.package_folder, "bin"))
+        rm(self, "*.la", self.package_folder, recursive=True)
+        # Remove symlinks to libexec/slapd
+        for path in self.package_path.joinpath("bin").glob("slap*"):
+            if path.is_symlink():
+                path.unlink()
+        # Remove irrelevant directories
+        for folder in ["var", "share", "etc", os.path.join("lib", "pkgconfig"), "res", "home", "libexec"]:
+            if os.path.exists(os.path.join(self.package_folder, folder)):
+                rmdir(self, os.path.join(self.package_folder, folder))
```

Comment:
```suggestion
        rm(self, "*.la", self.package_folder, recursive=True)
        for folder in ["var", "share", "etc", os.path.join("lib", "pkgconfig"), "res", "home"]:
            rmdir(self, os.path.join(self.package_folder, folder))
```

Let's keep the folder, much easier only adding it to cpp_info than adding more complexity.

Rmdir is fail safe, in case the folder does not exist, it does not raise error.